graphs/adapter-api/src/main/kotlin/com/anaplan/engineering/azuki/graphs/adapter/api/GraphActionFactory.kt

```diff
@@ -5,7 +5,21 @@ import com.anaplan.engineering.azuki.core.system.ActionFactory
 import com.anaplan.engineering.azuki.core.system.ParallelActionFactory
 import com.anaplan.engineering.azuki.core.system.UnsupportedAction
 
-interface GraphActionFactory<A: Action> : ActionFactory, ParallelActionFactory<A> {
+interface GraphActionFactory<A : Action> : ActionFactory, ParallelActionFactory<A> {
+
+    val undirected: undirectedActionFactory
+    val directed: directedActionFactory
+}
+
+interface undirectedActionFactory {
```

Comment:
🚫 Just like class names, interface names should be capitalised.
```suggestion
interface UndirectedActionFactory {
```